src/NuGet.Core/NuGet.Commands/RestoreCommand/ProjectRestoreCommand.cs

```diff
@@ -75,8 +75,11 @@ public ProjectRestoreCommand(ILogger logger, ProjectRestoreRequest request)
                 _request.MaxDegreeOfConcurrency,
                 token);
 
+            // The user global package folder is always ordered first.
+            var userPackageFolder = localRepositories.First();
```

Comment:
For the main API entry points it is that way. It might help here, but it will *always* be first. Splitting them up leads to constantly having to recombine them to loop over everything I've found.